### PR TITLE
bluetooth: controller: fixed problems in ISO-AL RX and included RX unit tests

### PR DESCRIPTION
**Pull request by niag-Oticon**
[*] Fixed setting of SDU production status when exiting the
    ISOAL_ERR_SPOOL state for framed PDU consumption when receiving the
    start of the next SDU
[*] Changed condition identifying a padding PDU to include padding PDUs
    with errors received after the end fragment is seen and expanded
    unit tests to cover new condition
[*] Changed definition of ISOAL_PDU_STATUS_xxx errors to match value of
    similar errors defined for the SDU status

Ztest based unit test of ISO adaptation layer.
Testing of unframed case, single SDU, recombined from
one and two PDUs.

### CODE REVIEW

**Review by mtpr-ot — DISMISSED** [review]
LGTM

File: tests/bluetooth/ctrl_isoal/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
+include_directories("./src")
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(bluetooth_ctrl_isoal)
+
+zephyr_library_include_directories(
+	${ZEPHYR_BASE}/subsys/bluetooth
```
Comment (SebastianBoe) [inline tests/bluetooth/ctrl_isoal/CMakeLists.txt:10]: Isn't this already in the include path?

at least it looks that way from

zephyr/subsys/bluetooth/CMakeLists.txt

please update other tests as well if they have the same problem

File: tests/bluetooth/ctrl_isoal/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
+include_directories("./src")
```
Comment (SebastianBoe) [inline tests/bluetooth/ctrl_isoal/CMakeLists.txt:4]: You don't hvae any header files in src.

Is someone #including main.c ?

**Review by SebastianBoe — CHANGES_REQUESTED** [review]
cmake changes requested

File: subsys/bluetooth/controller/ll_sw/isoal.c [hunk 3]
```diff
@@ -451,9 +452,17 @@ static isoal_status_t isoal_rx_unframed_consume(struct isoal_sink *sink,
...
 	next_state = ISOAL_START;
 
+	/* If status is not ISOAL_PDU_STATUS_VALID, length and LLID cannot be trusted */
 	llid = pdu_meta->pdu->ll_id;
 	pdu_err = (pdu_meta->meta->status != ISOAL_PDU_STATUS_VALID);
-	pdu_padding = (pdu_meta->pdu->length == 0) && (llid == PDU_BIS_LLID_START_CONTINUE);
+	length = pdu_err ? 0 : pdu_meta->pdu->length;
+	/* A zero length PDU with LLID 0b01 (PDU_BIS_LLID_START_CONTINUE) would be a padding PDU.
+	 * However if there are errors in the PDU, it could be an incorrectly receive non-padding
+	 * PDU. Therefore only consider a PDU with erros as padding if received after the end
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/isoal.c:461]: error
```suggestion
	 * PDU. Therefore only consider a PDU with error as padding if received after the end
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/isoal.c (hunk 3 above)
Comment (niag-Oticon) [reply subsys/bluetooth/controller/ll_sw/isoal.c]: Done.

File: tests/bluetooth/ctrl_isoal/CMakeLists.txt (hunk 2 above)
Comment (niag-Oticon) [reply tests/bluetooth/ctrl_isoal/CMakeLists.txt]: Removed. Done.

File: tests/bluetooth/ctrl_isoal/CMakeLists.txt (hunk 1 above)
Comment (niag-Oticon) [reply tests/bluetooth/ctrl_isoal/CMakeLists.txt:9]: The build fails if this is removed. It does not appear to be in the include path.

**Review by SebastianBoe — APPROVED** [review]
CMake changes are OK

**Review by cvinayak — APPROVED** [review]

**Review by mtpr-ot — APPROVED** [review]
LGTM

**Pull request merged**